scripts/ffcv/create_ffcv_datasets.py

```diff
@@ -0,0 +1,132 @@
+                        help=textwrap.dedent("""\
+                                WebDataset to use. Default: s3://mosaicml-internal-dataset-cifar10"""))
+    parser.add_argument("--split",
+                        type=str,
+                        default="train",
+                        choices=["train", "val"],
+                        help=textwrap.dedent("""\
+                                Split to use. Default: train"""))
+
+    parser.add_argument("--datadir",
+                        type=str,
+                        default=None,
+                        help=textwrap.dedent("""\
+                                Location of the dataset. Default: None"""))
+
+    parser.add_argument("--download",
+                        type=bool,
+                        default=False,
+                        help=textwrap.dedent("""\
+                                Download the dataset if possible. Default: False"""))
+
+    parser.add_argument("--write_path",
+                        type=str,
+                        default=None,
+                        help=textwrap.dedent("""\
+                                File path to use for writing the dataset. Default: /tmp/<dataset>_<split>.ffcv"""))
+
+    parser.add_argument("--write_mode",
+                        type=str,
+                        default="proportion",
+                        choices=["raw", "jpg", "smart", "proportion"],
+                        help=textwrap.dedent("""\
+                                Write mode to use. raw is uint8 values, jpg is jpeg compressed images, smart is
+                                compressing based on image size and proportion is according to the given
+                                compress_probability. Default: proportion"""))
+
+    parser.add_argument("--max_resolution", type=int, default=500, help="Max resoultion for images.")
+
+    parser.add_argument("--num_workers", type=int, default=16, help="Number of workers to use.")
+
+    parser.add_argument("--chunk_size", type=int, default=100, help="Chunk size to use.")
+
+    parser.add_argument("--jpeg_quality", type=int, default=90, help="Quality of jpeg.")
+
+    parser.add_argument("--subset", type=int, default=-1, help="Only use a subset f dataset.")
+
+    parser.add_argument("--compress_probability",
+                        type=float,
+                        required=False,
+                        default=0.50,
+                        help="Compress the given fraction of images to jpeg while writing the ffcv dataset.")
+    return parser
+
+
+def parse_args():
+    parser = get_parser()
+
+    args = parser.parse_args()
+
+    if args.datadir is not None:
+        log.info(f"Will read from local directory: {args.datadir}.")
+    else:
+        if args.remote.startswith('s3://'):
+            log.info(f"Will read from webdataset: {args.remote}.")
+        else:
+            raise ValueError(f'Unsupported webdataset location: {args.remote}.')
```

Comment:
Local filesystem remotes would be nice to have but not priority

Reply:
Will add these later if needed.